### ISSUE TITLE
[FIX] Autoscoping for snippet styles

### ISSUE THREAD

**Issue by atomiks** [30-seconds/30-seconds-of-css#102]
The current naive solution was to use a "namespace" (`.snippet-demo`) for the demo code, but it's arduous to type and repeats the main visible code. We should just use the code inside the main code blocks, and when building, generate a custom scope for the styles so they don't conflict with other snippets or the global styles. 

This means the demo should match the code exactly, which is better for the CodePen button. (Some don't match currently - as some only specify the minimum number of properties for the effect, but I think that causes confusion.)

**Comment by skatcat31** [comment]
I may be misunderstanding you, but won't generating a custom scope for the styles just repeat the code itself anyways? Or are you suggesting we move the namespacing chore to the build script?

**Comment by atomiks** [comment]
Example: [`circle.md`](https://raw.githubusercontent.com/30-seconds/30-seconds-of-css/master/snippets/circle.md)

The code that is actually displayed on the website is within `<style>`, notice, it's namespaced with `.snippet-demo__` so the styles don't conflict with the global ones or other snippets.

I'm suggesting to autoscope the selectors during the build script and forget about manually creating a demo -- the demo is created via the build script using the actual code.

#### Current

Snippet creator manually creates the demo, repeating code

```html
#### Demo

<div class="snippet-demo">
  <div class="snippet-demo__circle"></div>
</div>

<style>
.snippet-demo__circle {
  border-radius: 50%;
  width: 2rem;
  height: 2rem;
  background: #333;
}
</style>
```

#### New

The build script does it using the actual code within the blocks automatically.

```html
#### Demo
<!-- Leave this blank, the build script will generate the demo -->
```

Generated demo

```html
<div class="snippet-demo" data-scope="circle">
  <div class="circle"></div>
</div>

<!-- Not sure how to do this reliably -->
<style>
[data-scope="circle"] .circle {
  border-radius: 50%;
  width: 2rem;
  height: 2rem;
  background: #333;
}
</style>
```

Any suggestions on how to do this efficiently would be nice.

**Comment by skatcat31** [comment]
prepend any CSS selector with a data scope selector? The problem is that to understand this, the design needs to account for complex cases with several classes in the code. It may be better to generate a SCSS or .less string and pass it to the parse for output because then you merely need to wrap the style with the namespace selector and it'll do it for you?

**Comment by atomiks** [comment]
> It may be better to generate a SCSS or .less string and pass it to the parse for output because then you merely need to wrap the style with the namespace selector and it should do it for you.

Yeah I was thinking this. I thought a smart parser would be needed because my quick tries with regex were failing.

But with SCSS you can just stick everything inside a single block and it works. It shouldn't be too hard to convert the SCSS to CSS with the script.

**Comment by skatcat31** [comment]
either that or less, both will namespace inside the single block very similarly. It should be pretty easy and regex should be minimal. Gotta love those build tools right?

**Comment by atomiks** [comment]
Resolved in #103
